app/controllers/image_controller.rb

```diff
@@ -268,7 +230,74 @@ def prev_image
     redirect_to_next_object(:prev, Image, params[:id].to_s)
   end
 
+  private
+
+  def image_exists
+    @image = find_or_goto_index(Image, params[:id].to_s)
+  end
+
+  def set_default_size
+    # Decide which size to display.
+    @default_size = @user ? @user.image_size : :medium
+    @size = params[:size].blank? ? @default_size : params[:size].to_sym
+
+    # Maybe make this size the default image size for this user.
+    unless @user && (@default_size != @size) && (params[:make_default] == "1")
+      return
+    end
+
+    @user.image_size = @size
+    @user.save_without_our_callbacks
+    @default_size = @size
+  end
+
+  def set_image_query_params
+    obs = params[:obs]
+    # The outer search on observation won't be saved for robots, so no sense
+    # in bothering with any of this.
+    return unless obs.present? && obs.to_s.match(/^\d+$/) && !browser.bot?
+
+    obs_query = find_or_create_query(:Observation)
+    obs_query.current = obs
+    img_query = create_query(:Image, :inside_observation,
+                             observation: obs, outer: obs_query)
+    query_params_set(img_query)
+  end
+
+  def maybe_cast_user_vote
```

Comment:
```suggestion
  def cast_vote
```
See @pellaea's comment re `maybe_update_projects`

Reply:
Done